### PR TITLE
[create-sitecore-jss] New initializer journey for Angular

### PR DESCRIPTION
**Pull request by art-alexeyenko**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [ ] Upgrade guide entry added

## Description / Motivation
Angular will have XMC-only logic. This PR makes preparations for the new create-sitecore-jss init journey for Angular.

## Testing Details
- [ ] Unit Test Added
- [ ] Manual Test/Other (Please elaborate)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/create-sitecore-jss/src/initializers/angular/prompts.ts [hunk 1]
```diff
@@ -5,6 +5,31 @@ import {
...
-export const prompts = [...clientAppPrompts, ...styleguidePrompts];
+export const prompts = [
+  ...clientAppPrompts,
+  {
+    type: 'confirm',
+    name: 'xmcloud',
+    message: 'Are you building for Sitecore XM Cloud?',
+    default: false,
+    when: (answers: AngularAnswer): boolean => {
+      // don't prompt if --yes or nextjs-xmcloud template was specified
```
Comment (sc-addypathania) [inline packages/create-sitecore-jss/src/initializers/angular/prompts.ts:21]: should be angular-xmcloud?

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,9 @@ Our versioning strategy is as follows:
 
 ### 🎉 New Features & Improvements
 
+* `[create-sitecore-jss]``[angular]``[sitecore-jss-angular]` Introduce support for XM Cloud for Angular apps
+  * `[create-sitecore-jss]` Rework Angular initializer ([#1845](https://github.com/Sitecore/jss/pull/1845))
+
```
Comment (yavorsk) [inline CHANGELOG.md:24]: Change log for angular would probably follow a bit different format, but that can be resolved when merging with https://github.com/Sitecore/jss/pull/1838

**Review by yavorsk — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (art-alexeyenko) [reply CHANGELOG.md]: @yavorsk please check the updated entry. Should be better I think?

**Review by sc-addypathania — APPROVED** [review]
Looks good.

**Pull request merged**